core_lib/src/tool/basetool.h

```diff
@@ -50,6 +50,11 @@ class Properties
     int   fillMode = 0;
     int   stabilizerLevel = 0;
     qreal tolerance = 0;
+    bool toleranceEnabled = false;
+    int bucketFillExpand = 0;
+    bool bucketFillExpandEnabled = 0;
+    int bucketFillToLayerMode = 0;
+    int bucketFillReferenceMode = 0;
```

Comment:
Looks like all these newly added modes are exclusively for the bucket tool. I wonder if these variables could be moved to somewhere else better? And use `enum` instead?

Reply:
Yup I considered that too, maybe we should do that. I simply added it there since we have most of the tool properties in basetool.